nomad/structs/csi.go

```diff
@@ -880,6 +867,20 @@ func (v *CSIVolume) Merge(other *CSIVolume) error {
 	return errs.ErrorOrNil()
 }
 
+// MergeWhileInUse updates fields that may be mutated while the volume is in use.
+func (v *CSIVolume) MergeWhileInUse(other *CSIVolume) error {
+	if v == nil || other == nil {
+		return nil // TODO: error?
```

Comment:
It's probably a programmer's error in the call sites we have, but b/c it's being done in `nomad/structs` we don't have a lot of guarantees someone won't misuse it elsewhere later.

Reply:
yeah I wanted to nil check at least, but should I make it an explicit error, like "this is probably a bug" instead of just returning nil?

Reply:
Make sense